plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/ProductVariationTemplate.php

```diff
@@ -100,6 +106,23 @@ private function add_group_blocks() {
 	 */
 	private function add_general_group_blocks() {
 		$general_group = $this->get_group_by_id( $this::GROUP_IDS['GENERAL'] );
+		$product_id    = ProductEditorHelper::get_parsed_route()['product_id'];
```

Comment:
After discussing this with @mdperez86, we agreed that this would be checked only on the package side. I addressed the change in the commit [9a4e867](https://github.com/woocommerce/woocommerce/pull/40679/commits/9a4e8679852633adc7a54ce8c9e4b55048156235).